File: Tempo_odnc/Bubblesort.py
import time
import psutil
import os

def bubble_sort(lista):
    n = len(lista)
    for i in range(n):
        for j in range(0, n-i-1):
            if lista[j] > lista[j+1]:
                lista[j], lista[j+1] = lista[j+1], lista[j]
    return lista

def ler_numeros(nome_arquivo):
    if not os.path.isfile(nome_arquivo):
        raise FileNotFoundError(f"O arquivo {nome_arquivo} não existe.")
    with open(nome_arquivo, 'r') as arquivo:
        return [int(linha.strip()) for linha in arquivo]

def escrever_numeros(nome_arquivo, numeros):
    with open(nome_arquivo, 'w') as arquivo:
        for numero in numeros:
            arquivo.write(f"{numero}\n")

def medir_memoria():
    processo = psutil.Process(os.getpid())
    return processo.memory_info().rss / 1024  

if __name__ == "__main__":
    arquivo_entrada = 'arq(1).txt'  
    arquivo_saida = 'arq-saida.txt'
    
    try:
        numeros = ler_numeros(arquivo_entrada)
    except FileNotFoundError as e:
        print(e)
        exit(1)
    
    tempo_inicio = time.time()
    numeros_ordenados = bubble_sort(numeros)
    tempo_fim = time.time()
    
    escrever_numeros(arquivo_saida, numeros_ordenados)
    
    tempo_gasto = (tempo_fim - tempo_inicio) * 1000  # em ms
    memoria_usada = medir_memoria()  # em KB
    
    print(f"Tempo gasto: {tempo_gasto:.2f} ms")
    print(f"Memória usada: {memoria_usada:.2f} KB")

File: Tempo_odnc/BubbleSort.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;

 public class BubbleSort {

    public static void main(String[] args) {
        String arquivoEntrada = "arq(1).txt";
        String arquivoSaida = "arq-saida.txt";

        List<Integer> numeros = lerNumeros(arquivoEntrada);
        if (numeros == null) {
            System.out.println("Erro ao ler o arquivo de entrada.");
            return;
        }

        long tempoInicio = System.currentTimeMillis();
        List<Integer> numerosOrdenados = bubbleSort(numeros);
        long tempoFim = System.currentTimeMillis();

        escreverNumeros(arquivoSaida, numerosOrdenados);

        long tempoGasto = tempoFim - tempoInicio;  
        long memoriaUsada = medirMemoria();  

        System.out.printf("Tempo gasto: %d ms%n", tempoGasto);
        System.out.printf("Memória usada: %d KB%n", memoriaUsada);
    }

    public static List<Integer> bubbleSort(List<Integer> lista) {
        int n = lista.size();
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (lista.get(j) > lista.get(j + 1)) {
                    int temp = lista.get(j);
                    lista.set(j, lista.get(j + 1));
                    lista.set(j + 1, temp);
                }
            }
        }
        return lista;
    }

    public static List<Integer> lerNumeros(String nomeArquivo) {
        List<Integer> numeros = new ArrayList<>();
        try (BufferedReader br = new BufferedReader(new FileReader(nomeArquivo))) {
            String linha;
            while ((linha = br.readLine()) != null) {
                numeros.add(Integer.parseInt(linha.trim()));
            }
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }
        return numeros;
    }

    public static void escreverNumeros(String nomeArquivo, List<Integer> numeros) {
        try (BufferedWriter bw = new BufferedWriter(new FileWriter(nomeArquivo))) {
            for (int numero : numeros) {
                bw.write(String.valueOf(numero));
                bw.newLine();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static long medirMemoria() {
        Runtime runtime = Runtime.getRuntime();
        runtime.gc();  
        long memoriaUsada = (runtime.totalMemory() - runtime.freeMemory()) / 1024;  
        return memoriaUsada;
    }
}
